pkg/natdiscovery/remote_endpoint.go

```diff
@@ -86,17 +102,26 @@ func (rn *remoteEndpointNAT) sinceLastTransition() time.Duration {
 }
 
 func (rn *remoteEndpointNAT) hasTimedOut() bool {
-	return time.Since(rn.started) > toDuration(&totalTimeout)
+	return time.Since(rn.started) > rn.timeout
 }
 
 func (rn *remoteEndpointNAT) useLegacyNATSettings() {
-	rn.useNAT = rn.endpoint.Spec.NATEnabled
-	if rn.endpoint.Spec.NATEnabled {
+	switch {
+	case rn.usingLoadBalancer:
```

Comment:
We are giving priority to LoadBalancer first, followed by Spec.NATEnabled and finally falling back to default. Since we are using a switch it might not be immediately evident for people looking at the code. Can you please add a small comment.

Reply:
But that's an inner characteristic of switch in go. I think we don't need to explain go to the reader, I don't know ':D  let's see what other reviewers think of this.

@sridhargaddam thanks a lot for reviewing